walkthroughs/howto-k8s-ingress-gateway/README.md

```diff
@@ -0,0 +1,118 @@
+# Configuring Ingress Gateway
+
+In this walkthrough, we'll configure an Ingress Gateway in our existing example color app but with a VirtualGateway resource instead of VirtualNode for ingress traffic.
+
+A virtual gateway allows resources outside your mesh to communicate to resources that are inside your mesh. The virtual gateway represents an Envoy proxy running in an Amazon ECS, in a Kubernetes service, or on an Amazon EC2 instance. Unlike a virtual node, which represents a proxy running with an application, a virtual gateway represents the proxy deployed by itself.
+
+## Prerequisites
+This feature is currently only available in App Mesh preview and will work with App Mesh manager [here](https://github.com/aws/eks-charts/tree/master/stable/appmesh-manager)
+
+This example requires [aws-app-mesh-controller-for-k8s](https://github.com/aws/aws-app-mesh-controller-for-k8s) version [>=v1.0.0](https://github.com/aws/aws-app-mesh-controller-for-k8s/blob/master/CHANGELOG.md). Run the following to check the version of controller you are running.
+```
+$ kubectl get deployment -n appmesh-system appmesh-manager -o json | jq -r ".spec.template.spec.containers[].image" | cut -f2 -d ':'|tail -n1
+```
+
+## Setup
+
+1. Clone this repository and navigate to the walkthrough/howto-k8s-ingress-gateway folder, all commands will be ran from this location
+2. **Your** account id:
+
+    export AWS_ACCOUNT_ID=<your_account_id>
+
+3. **Region** e.g. us-west-2
+
+    export AWS_DEFAULT_REGION=us-west-2
+
+4. **ENVOY_IMAGE** environment variable is set to App Mesh Envoy, see https://docs.aws.amazon.com/app-mesh/latest/userguide/envoy.html
+
+    export ENVOY_IMAGE=...
+
+5. Deploy
+```.
+    ./deploy.sh
+```
+
+## Use Ingress gateway
+
+There are two GatewayRoutes setup in this example: 1) `gateway-route-headers` 2) `gateway-route-paths`.
+`gateway-route-headers` will route traffic to VirtualService `color-headers` and `gateway-route-paths` will route traffic to VirtualService `color-paths`
+
+VirtualService `color-headers` uses a VirtualRouter to match HTTP headers to choose the backend VirtualNode. VirtualService `color-paths` uses HTTP path prefixes to choose backend VirtualNode
+
+Let's look at the VirtualGateway deployed:
+
+```
+kubectl get virtualgateway -n howto-k8s-ingress-gateway                                         
+NAME         ARN                                                                                                                                 AGE
+ingress-gw   arn:aws:appmesh-preview:us-west-2:112233333455:mesh/howto-k8s-ingress-gateway/virtualGateway/ingress-gw_howto-k8s-ingress-gateway   113s
+```
```

Comment:
Shouldn't there also be an NLB deployed in front of the gateway? Need to demonstrate that as well.

Reply:
+1, will add